packages/framework-provider-local/test/library/scheduled-adapter.test.ts

```diff
@@ -0,0 +1,44 @@
+import { BoosterConfig } from '@boostercloud/framework-types'
+import { expect } from '../expect'
+import { fake, SinonStub, stub } from 'sinon'
+import { rawScheduledInputToEnvelope } from '../../src/library/scheduled-adapter'
+import { describe } from 'mocha'
+
+describe('scheduled-adapter', () => {
+  let mockConfig: BoosterConfig
+  let loggerDebugStub: SinonStub
+
+  beforeEach(() => {
+    mockConfig = new BoosterConfig('test')
+    mockConfig.appName = 'nuke-button'
+
+    loggerDebugStub = stub()
+
+    mockConfig.logger = {
+      info: fake(),
+      warn: fake(),
+      error: fake(),
+      debug: loggerDebugStub,
+    }
+  })
+
+  describe('rawScheduledInputToEnvelope', () => {
+    ;[{ typeName: '' }, { typeName: undefined }].forEach(function (arg) {
```

Comment:
Is there any way to make this more readable? 

I really believe that the unit tests are a lot of times the entry points for possible contributors, or even users that are trying to figure out how does the framework work under the hood.

If we keep the code as readable as possible and keep the barrier low here, it can help a lot of people :)

Reply:
Just to confirm, you're referring specifically to this line only, right?
`;[{ typeName: '' }, { typeName: undefined }].forEach(function (arg) {`

I understand your point, it can be a bit unwieldy for a first-time reader. In that case, I see 2 options.

Option 1 - decompose the expression:
```
const emptyScheduledCommandEnvelope = { typeName: '' }
const undefinedScheduledCommandEnvelope = { typeName: undefined }

const testCases = [emptyScheduledCommandEnvelope, undefinedScheduledCommandEnvelope]
testCases.forEach(function (arg) {
(...)
```

Option 2 - decompose the test itself into 2:
```
it('should throw an error when typeName is empty', async () => {
        const emptyScheduledCommandEnvelope = { typeName: '' }
        await expect(rawScheduledInputToEnvelope(mockConfig, emptyScheduledCommandEnvelope)).to.eventually.be.rejectedWith(
          new Error(
            `typeName is not defined or empty, scheduled command envelope should have the structure {typeName: string }, but you gave ${JSON.stringify(
              arg
            )}`
          ).message
        )
      })

it('should throw an error when typeName is undefined', async () => {
        const undefinedScheduledCommandEnvelope = { typeName: undefined }
        await expect(rawScheduledInputToEnvelope(mockConfig, emptyScheduledCommandEnvelope)).to.eventually.be.rejectedWith(
          new Error(
            `typeName is not defined or empty, scheduled command envelope should have the structure {typeName: string }, but you gave ${JSON.stringify(
              arg
            )}`
          ).message
        )
      })
```

I would personally go with option 1 because I see it as 2 very similar forms of invalid input, but what would you prefer?

Reply:
Although I usually prefer writing highly abstract code that removes duplication, **for tests** I believe it is much better if every test can be read as a single case that makes clear the point of it.

So for this case, I'd go with option 2